mockall_derive/src/mock_function.rs

```diff
@@ -986,15 +986,22 @@ impl<'a> ToTokens for Common<'a> {
 
             impl #ig Drop for Common #tg #wc {
                 fn drop(&mut self) {
-                    if !::std::thread::panicking() && !self.times.is_satisfied()
-                    {
-                        let desc = std::format!(
-                            "{}", self.matcher.lock().unwrap());
-                        panic!("{}: Expectation({}) called {} time(s) which is fewer than expected {}",
-                               #funcname,
-                               desc,
-                               self.times.count(),
-                               self.times.minimum());
+                    if !::std::thread::panicking() {
+                        match self.times.is_satisfied() {
+                            ::mockall::ExpectedCalls::SatisfiedOnlyUpperBound => {
+                                let desc = std::format!(
+                                    "{}", self.matcher.lock().unwrap());
+                                panic!("{}: Expectation({}) called {} time(s) which is fewer than expected {}",
+                                    #funcname,
+                                    desc,
+                                    self.times.count(),
+                                    self.times.minimum());
+                            },
+                            ::mockall::ExpectedCalls::SatisfiedOnlyLowerBound => {
+                                panic!("Exceeded number of expected calls panic propagation");
```

Comment:
What's this about panic propagation?  A better message would be:
```suggestion
                                panic!("{}: Expectation({}) called {} time(s) which is more than the permitted {}",
                                    #funcname,
                                    desc,
                                    self.times.count(),
                                    self.times.maximum() - 1);
```

Reply:
This message is already printed by the panic in the call() method.
So in a subthread scenario there will be 2 equal panic messages.
I wanted to differentiate the second one.

Or did you expected to remove completely the panic in the call() method? And have only one in drop()?

Reply:
Will it be?  The second message will only be printed if for some reason the user is ignoring panics.  In that case, he'll need all the info in the second one.

Reply:
when calls happen in subthread and are more than expected fitst there is the panic from the check in call() method.
but it's ignored by main thread.
then there is the panic from the check in drop() method that makes the test fail.

but even if the messages are the same and are printed 2 times in this scenario it's not a problem, so I changed the message as suggested.